sdk/storage/Azure.Storage.DataMovement/src/CommitChunkHandler.cs

```diff
@@ -70,7 +83,21 @@ public struct Behaviors
             // Set expected length to perform commit task
             _expectedLength = expectedLength;
 
+            // Create channel of finished Stage Chunk Args to update the bytesTransferred
+            // and for ending tasks like commit block.
+            // The size of the channel should never exceed 50k (limit on blocks in a block blob).
+            // and that's in the worst case that we never read from the channel and had a maximum chunk blob.
+            _stageChunkChannel = Channel.CreateUnbounded<StageChunkEventArgs>(
+                new UnboundedChannelOptions()
+                {
+                    // Single reader is required as we can only read and write to bytesTransferred value
+                    SingleReader = true,
+                });
+            _cancellationTokenSource = new CancellationTokenSource();
+            _processStageChunkEvents = Task.Run(() => NotifyOfPendingStageChunkEvents());
+
```

Comment:
Will cherry pick these changes out and put in a different PR for future review and testing.

https://github.com/Azure/azure-sdk-for-net/issues/32854